massa-client/src/tests/scenarios.rs

```diff
@@ -1,40 +1,63 @@
+use anyhow::{bail, Result};
 use assert_cmd::Command;
+use massa_models::api::NodeStatus;
+use serde::de::DeserializeOwned;
 use serial_test::serial;
-use std::thread;
 
-#[tokio::test]
-#[serial]
-async fn test_if_exit_gracefully() {
-    let mut cmd = Command::cargo_bin("massa-client").unwrap();
-    cmd.arg("exit").assert().success();
+const TIMEOUT: u64 = 5;
+const TENTATIVES: u64 = 5;
+
+async fn run_client_cmd<T: DeserializeOwned>(cmd: &str) -> Result<T> {
+    let mut massa_client = Command::cargo_bin("massa-client")?;
+    for _ in 0..TENTATIVES {
+        let mut output = std::str::from_utf8(
+            &massa_client
+                .args(&[cmd, "--json"])
+                .assert()
+                .get_output()
+                .stdout,
+        )?
+        .to_string();
+        // TODO: compare with tcp connect error as a JSON object?
+        if !output.contains("tcp connect error") {
+            // TODO: serde fail to deserialize `()` type from `""` string
```

Comment:
Where does that happen ?

Reply:
https://github.com/massalabs/massa/runs/4504646245?check_suite_focus=true#step:6:54